src/domain/valueObject/secureAccessKeyContent.go

```diff
@@ -0,0 +1,49 @@
+package valueObject
+
+import (
+	"errors"
+	"regexp"
+	"strings"
+
+	voHelper "github.com/goinfinite/os/src/domain/valueObject/helper"
+)
+
+const secureAccessKeyContentRegex string = `^(?:ssh-rsa) (?:[\w\/\+\=]+)(?: [\w@\-_]{6,32})?$`
+
+type SecureAccessKeyContent string
+
+func NewSecureAccessKeyContent(
+	value interface{},
+) (keyContent SecureAccessKeyContent, err error) {
+	stringValue, err := voHelper.InterfaceToString(value)
+	if err != nil {
+		return keyContent, errors.New("SecureAccessKeyContentMustBeString")
+	}
+
+	re := regexp.MustCompile(secureAccessKeyContentRegex)
+	if !re.MatchString(stringValue) {
+		return keyContent, errors.New("InvalidSecureAccessKeyContent")
+	}
+
+	return SecureAccessKeyContent(stringValue), nil
+}
+
+func (vo SecureAccessKeyContent) String() string {
+	return string(vo)
+}
+
+func (vo SecureAccessKeyContent) ReadWithoutKeyName() string {
+	keyContentParts := strings.Split(string(vo), " ")
+	return keyContentParts[0] + " " + keyContentParts[1]
+}
```

Comment:
I'm pretty sure not all keys have a second part, some will not have what you call key name. It would be best to have a fallback for this cases.

Reply:
Position [0] represents the encoding type, such as `ssh-rsa` or `ssh-ed25519`. Position [1] is the key content itself. Only position [2] corresponds to the key name, which is optional.